ui/docs/ember-engines.md

```diff
@@ -0,0 +1,158 @@
+# [Ember Engines](https://ember-engines.com/docs)
+
+This is a quickstart guide inspired by [ember engine quickstart](https://ember-engines.com/docs/quickstart) on how to set up an ember engine in Vault!
+
+## Create a new in-repo engine:
+
+`ember g in-repo-engine <engine-name>`
+
+_This blueprint in-repo engine command will add a new folder `lib/<engine-name>` and add the engine to our main app’s `package.json`_
+
+## Engine’s package.json:
+
+```json
+{
+  "name": "<engine-name>",
+
+  "dependencies": {
+    "ember-cli-htmlbars": "*",
+    "ember-cli-babel": "*"
+  },
+
+  "ember-addon": {
+    "paths": ["../core"]
+  }
+}
+```
+
+For our application, we want to include the **ember-addon** path `../core`
+
+By adding this **ember-addon** path, we are able to share elements between your in-repo addon and the Vault application[^1].
+
+## Configure your Engine
+
+In the engine’s `index.js` file:
+
+```jsx
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: MPL-2.0
+ */
+
+/* eslint-disable node/no-extraneous-require */
+const { buildEngine } = require('ember-engines/lib/engine-addon');
+
+module.exports = buildEngine({
+  name: '<engine-name>',
+  lazyLoading: {
+    enabled: false,
+  },
+  isDevelopingAddon() {
+    return true;
+  },
+});
+```
+
+Within your Engine’s `app/config/environment.js` file:
+
+```jsx
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: MPL-2.0
+ */
+
+// config/environment.js
+
+'use strict';
+
+module.exports = function (environment) {
+  const ENV = {
+    modulePrefix: '<engine-name>',
+    environment: environment,
+  };
+
+  return ENV;
+};
+```
+
+Within your Engine’s `addon/engine.js` file:
+
+```jsx
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: MPL-2.0
+ */
+
+import Engine from '@ember/engine';
+
+import loadInitializers from 'ember-load-initializers';
+import Resolver from 'ember-resolver';
+
+import config from './config/environment';
+
+const { modulePrefix } = config;
+
+export default class <EngineName>Engine extends Engine {
+  modulePrefix = modulePrefix;
+  Resolver = Resolver;
+  dependencies = {
+    services: ['router', 'store', 'secret-mount-path', 'flash-messages'],
+    externalRoutes: ['secrets'],
```

Comment:
Maybe a comment about why you want an external route. If you link to a route from within your engine and it's to a page outside your engine, you'll need to add it to this array. Also, this is tied to the app.js file you mention later on. Where you list the string secrets here it's because it's defined the route 'secrets' in the app.js file where that string points to..

Reply:
Awesome idea! I'll add an explanation for external route!